File: src/praktikum3/kegiatan2String/CoffeeQueue.java
package praktikum3.kegiatan2String;

import java.util.Scanner;

public class CoffeeQueue {

    public FirstLastLink theQueue;

    public CoffeeQueue() {
        theQueue = new FirstLastLink();
    }

    public void enqueue(String data) {
        theQueue.insertLast(data);
    }

    public String dequeue() {
        return theQueue.deleteFirst();
    }

    public void peek() {
        System.out.println("Top queue : " + theQueue.first.dData);
    }

    public boolean isEmpty() {
        if (theQueue.first == null) {
            return true;
        } else {
            return false;
        }
    }

    public void size() {
        int temp = 0;
        Link current = theQueue.first;

        while (current != null) {
            temp++;
            current = current.next;
        }
        System.out.println("Size : " + temp + "\n");
    }

    public void displayQueue() {
        System.out.print("Queue [Head -> Tail] : ");
        theQueue.displayLink();
    }

    public static void main(String[] args) {
        CoffeeQueue coffeeQueue = new CoffeeQueue();

        // coffeeQueue.enqueue("Matcha");
        // coffeeQueue.enqueue("Cappucino");
        // coffeeQueue.displayQueue();

        // coffeeQueue.enqueue("Americano");
        // coffeeQueue.enqueue("Espresso");
        // coffeeQueue.displayQueue();

        // coffeeQueue.dequeue();
        // coffeeQueue.displayQueue();

        // coffeeQueue.peek();
        // coffeeQueue.size();

        // System.out.println("Empty : " + coffeeQueue.isEmpty());

        // menu
        Scanner sc = new Scanner(System.in);
        int pilihan = 0;
        String input = "";

        do {
            System.out.println("Coffee Shop");
            System.out.println("Menu : ");
            System.out.println("1. Enqueue");
            System.out.println("2. Dequeue");
            System.out.println("3. Peek");
            System.out.println("4. Size");
            System.out.println("5. Display");
            System.out.println("6. Empty");
            System.out.println("7. Exit");
            System.out.print("Pilihan : ");
            pilihan = sc.nextInt();
            sc.nextLine();

            switch (pilihan) {
                case 1:
                    System.out.print("Masukkan input : ");
                    input = sc.nextLine();
                    coffeeQueue.enqueue(input);
                    break;
                case 2:
                    coffeeQueue.dequeue();
                    break;
                case 3:
                    coffeeQueue.peek();
                    break;
                case 4:
                    coffeeQueue.size();
                    break;
                case 5:
                    coffeeQueue.displayQueue();
                    break;
                case 6:
                    System.out.println("Empty : " + coffeeQueue.isEmpty() + "\n");
                    break;
                case 7:
                    System.out.println("Terima kasih");
                    break;
                default:
                    System.out.println("Pilihan tidak tersedia");
                    break;
            }
        } while (pilihan != 7);

        sc.close();
    }
}

File: src/praktikum3/kegiatan2String/CoffeeQueue.py
__author__      = "rizkyhaksono"
__copyright__   = "Copyright 2023, Malang"

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class CoffeeQueue:
    def __init__(self): 
        self.head = None

    def isEmpty(self):
        return self.head == None

    def enqueue(self, data):
        if self.head == None:
            self.head = Node(data)
        else:
            node = self.head
            while node.next != None:
                node = node.next
            node.next = Node(data)

    def dequeue(self):
        if self.head == None:
            return None
        else:
            popped = self.head.data
            self.head = self.head.next
            return popped

    def peek(self):
        if self.head == None:
            return None
        else:
            return self.head.data

    def show(self):
        if self.head == None:
            return None
        else:
            node = self.head
            while node != None:
                print(node.data, end = " ")
                node = node.next
            print()

    def size(self):
        node = self.head
        count = 0
        while node != None:
            count += 1
            node = node.next
        return count

if __name__ == "__main__":
    c = CoffeeQueue()

    # c.enqueue("Matcha")
    # c.enqueue("Cappucino")
    # print("Queue [Head -> Tail] : ", end = "")
    # c.show()

    # c.enqueue("Americano")
    # c.enqueue("Espresso")
    # print("Queue [Head -> Tail] : ", end = "")
    # c.show()
    
    # print("Remove: ", c.dequeue())
    # print("Queue [Head -> Tail] : ", end = "")
    # c.show()

    # print("Top queue : ", c.peek())
    # print("Queue [Head -> Tail] : ", end = "")
    # c.show()

    # print("Size : ", c.size())
    # print("Empty : ", c.isEmpty())

    # menu
    while True:
        print("\n===============================")
        print("Coffee Queue")
        print("===============================")
        print("1. Enqueue")
        print("2. Dequeue")
        print("3. Peek")
        print("4. Show")
        print("5. Size")
        print("6. Empty")
        print("7. Exit")
        print("===============================")
        menu = int(input(">> "))
        print("===============================")

        if menu == 1:
            data = input("Masukkan data: ")
            c.enqueue(data)
        elif menu == 2:
            print("Data yang dikeluarkan: ", c.dequeue())
        elif menu == 3:
            print("Data teratas: ", c.peek())
        elif menu == 4:
            print("Queue [Head -> Tail] : ", end = "")
            c.show()
        elif menu == 5:
            print("Size : ", c.size())
        elif menu == 6:
            print("Empty : ", c.isEmpty())
        elif menu == 7:
            break
        else:
            print("Menu tidak tersedia")

    print("Program selesai")